plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1553,7 +1554,7 @@ public void setTableProperties(ConnectorSession session, ConnectorTableHandle ta
         }
 
         try {
-            transaction.commitTransaction();
+            commitTransaction(transaction);
         }
         catch (RuntimeException e) {
             throw new TrinoException(ICEBERG_COMMIT_ERROR, "Failed to commit new table properties", e);
```

Comment:
if you're dealing with a TrinoException, just throw it.
Add another catch specific for `TrinoException`

Reply:
fixed